src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1457,13 +1460,25 @@ private IEnumerable<TypeInferenceInfo> InferTypeInRecursivePattern(RecursivePatt
                 // we have  { X: ... }.  The type of ... is whatever the type of 'X' is in its
                 // parent type.  So look up the parent type first, then find the X member in it
                 // and use that type.
-                if (child == subpattern.Pattern &&
-                    // PROTOTYPE(extended-property-patterns) ExpressionColon
-                    subpattern.NameColon != null)
+                if (child == subpattern.Pattern)
                 {
-                    var result = ArrayBuilder<TypeInferenceInfo>.GetInstance();
+                    if (subpattern.NameColon != null)
+                    {
+                        return InferTypeInSubpatternCore(subpattern.NameColon.Name);
+                    }
+                    else if (subpattern.ExpressionColon is { Expression: MemberAccessExpressionSyntax memberAccess })
+                    {
+                        return InferTypeInSubpatternCore(memberAccess.Name);
+                    }
```

Comment:
nit: You could probably just call `InferTypeInSubpatternCore(subpattern.ExpressionColon?.Expression)`
We've verified GetSymbolInfo works with a member access.

Reply:
```suggestion 
                    if (subpattern.ExpressionColon != null)
                    {
                        return InferTypeInSubpatternCore(subpattern.ExpressionColon.Expression);
                    }
```